src/ResourceManager/KeyVault/Commands.KeyVault/Commands/NewAzureKeyVaultCertificatePolicy.cs

```diff
@@ -27,130 +26,164 @@ namespace Microsoft.Azure.Commands.KeyVault
     /// </summary>
     [Cmdlet(VerbsCommon.New, CmdletNoun.AzureKeyVaultCertificatePolicy,
         SupportsShouldProcess = true,
+        DefaultParameterSetName = SubjectNameParameterSet,
         HelpUri = Constants.KeyVaultHelpUri)]
-    [OutputType(typeof(KeyVaultCertificatePolicy))]
+    [OutputType(typeof(PSKeyVaultCertificatePolicy))]
     public class NewAzureKeyVaultCertificatePolicy : KeyVaultCmdletBase
     {
+        #region Parameter Set Names
+
+        private const string SubjectNameParameterSet = "SubjectName";
+        private const string DNSNamesParameterSet = "DNSNames";
+
+        #endregion
+
         #region Input Parameter Definitions
 
+        /// <summary>
+        /// IssuerName
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   Position = 0,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specifies the name of the issuer for this certificate.")]
+        public string IssuerName { get; set; }
+
+        /// <summary>
+        /// SubjectName
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   Position = 1,
+                   ParameterSetName = SubjectNameParameterSet,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specifies the subject name of the certificate.")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = DNSNamesParameterSet,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specifies the subject name of the certificate.")]
+        public string SubjectName { get; set; }
+
+        /// <summary>
+        /// DnsNames
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   Position = 1,
+                   ParameterSetName = DNSNamesParameterSet,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specifies the DNS names in the certificate.")]
+        public List<string> DnsNames { get; set; }
```

Comment:
should rename as DnsName, with alias to old name DnsNames (and deprecate)

Reply:
Why? This parameter does take a list..

Reply:
@dragav it comes from the [_Strongly Encouraged Developer Guidelines_](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx) for PowerShell cmdlets:

_Avoid using plural names for parameters whose value is a single element. This includes parameters that take arrays or lists because the user might supply an array or list with only one element._

Reply:
I disagree. We don't have a parameter called "PermissionToSecrets" - plural is appropriate here as it conveys the fact that the certificate accepts an enumeration of DNS names. Having a suggestively named parameter outweighs a convention based solely on the parameter's type.

Reply:
@dragav I agree that this convention is silly, but it is one that we enforce for new changes across _all_ modules and one that has been brought up by users numerous times (_e.g._, [this issue](https://github.com/Azure/azure-powershell/issues/2919)).

Going back to the guidelines mentioned above, as long as the user can provide a _single_ object to the list/array parameter, than the parameter noun should be singular. If, however, you can do extra validation to ensure that the user _always_ provides more than one object, then having a plural noun is allowed:

_Plural parameter names should be used only in those cases where the value of the parameter is always a multiple-element value. In these cases, the cmdlet should verify that multiple elements are supplied, and the cmdlet should display a warning to the user if multiple elements are not supplied._